vector/src/test/java/im/vector/app/features/location/live/map/GetListOfUserLiveLocationUseCaseTest.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.location.live.map
+
+import androidx.lifecycle.asFlow
+import com.airbnb.mvrx.test.MvRxTestRule
+import im.vector.app.features.location.LocationData
+import im.vector.app.test.fakes.FakeSession
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.unmockkStatic
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.runTest
+import org.amshove.kluent.internal.assertEquals
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.matrix.android.sdk.api.session.room.model.livelocation.LiveLocationShareAggregatedSummary
+import org.matrix.android.sdk.api.session.room.model.message.MessageBeaconLocationDataContent
+
+class GetListOfUserLiveLocationUseCaseTest {
+
+    @get:Rule
+    val mvRxTestRule = MvRxTestRule()
```

Comment:
we should be able to avoid this rule in a non MvRx viewmodel test :crossed_fingers:

Reply:
In fact, you are right it is not needed in this particular useCase but I added it as extra security. If we decide to switch the Coroutine Dispatcher inside the implementation, it will not work. The rule sets the `TestCoroutineDispatcher` to avoid such problems. We could also define our own rule if we want to avoid using that `MvRx` rule.

Reply:
I would argue for manually injecting dispatchers rather than relying on a rule to override the global dispatchers (it's not possible to pass a dispatcher to the android `ViewModel/LiveData` which is why it's needed in some cases) 

we have [injectable dispatchers](https://github.com/vector-im/element-android/blob/develop/vector/src/main/java/im/vector/app/core/dispatchers/CoroutineDispatchers.kt) and [test dispatchers](https://github.com/vector-im/element-android/blob/develop/vector/src/test/java/im/vector/app/test/TestCoroutineDispatchers.kt) for this scenario